### PR TITLE
[stag] GOV-1233 Readme/Link CRUD Governance issue

### PR DESCRIPTION
**Pull request by nikhilbonte21**
https://linear.app/atlanproduct/issue/GOV-1233/readmelink-crud-governance-issue

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    private final AtlasTypeRegistry typeRegistry;
+    private final EntityGraphRetriever entityRetriever;
+
+    AbstractResourcePreProcessor(AtlasTypeRegistry typeRegistry,
+                              EntityGraphRetriever entityRetriever) {
+        this.typeRegistry = typeRegistry;
+        this.entityRetriever = entityRetriever;
+    }
+
+    void authorizeResourceUpdate(AtlasEntity resourceEntity, AtlasVertex ResourceVertex, String edgeLabel) throws AtlasBaseException {
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java:56]: Everything looks Good! Tested it working as expected
One pointers to add, regarding the code 

We can make `authorizeResourceDelete` and `authorizeResourceUpdate` funcction bit more crisp, lot of repetition  in authZ logic, can feel could be embedded in one function.

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/LinkPreProcessor.java [hunk 2]
```diff
@@ -1,55 +1,87 @@
...
 
+    private void processLinkCreate(AtlasEntity linkEntity) throws AtlasBaseException {
+        validateLinkAttribute(linkEntity, CREATE.name());
+    }
+
+    private void processLinkUpdate(AtlasEntity linkEntity, AtlasVertex vertex) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processLinkUpdate");
+
+        try {
+            validateLinkAttribute(linkEntity, UPDATE.name());
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/LinkPreProcessor.java:65]: Lets do authZ first

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java (hunk 1 above)
Comment (n5nk) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java:56]: +1 See if we can avoid DRY here.

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java [hunk 3]
```diff
@@ -581,7 +582,7 @@ public EntityMutationResponse deleteById(final String guid) throws AtlasBaseExce
         if (vertex != null) {
             AtlasEntityHeader entityHeader = entityRetriever.toAtlasEntityHeaderWithClassifications(vertex);
 
-            AtlasAuthorizationUtils.verifyAccess(new AtlasEntityAccessRequest(typeRegistry, AtlasPrivilege.ENTITY_DELETE, entityHeader), "delete entity: guid=", guid);
+            AtlasAuthorizationUtils.verifyDeleteEntityAccess(new AtlasEntityAccessRequest(typeRegistry, AtlasPrivilege.ENTITY_DELETE, entityHeader), "delete entity: guid=", guid);
```
Comment (n5nk) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java:585]: we are already calling the method as `verifyDeleteEntityAccess`, why do we need to pass `AtlasPrivilege.ENTITY_DELETE` again ? same with all others

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/ReadmePreProcessor.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            case CREATE:
+                processCreateReadme(entity);
+                break;
+            case UPDATE:
+                processUpdateReadme(entity, context.getVertex(entity.getGuid()));
+                break;
+        }
+    }
+
+    private void processCreateReadme(AtlasStruct struct) throws AtlasBaseException {
```
Comment (n5nk) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/ReadmePreProcessor.java:62]: if we are not throwing exceptions, we can remove those

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/LinkPreProcessor.java (hunk 2 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/LinkPreProcessor.java:65]: I think it is better to validate the link value before checking for authz

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java (hunk 3 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java:585]: Refactoring this

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/ReadmePreProcessor.java (hunk 4 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/ReadmePreProcessor.java:62]: Removing method & corresponding switch case

**Review by n5nk — APPROVED** [review]

**Pull request merged**